extensions/amp-form/0.1/amp-form.js

```diff
@@ -625,10 +693,14 @@ export class AmpForm {
 
   /**
    * Handles response redirect throught the AMP-Redirect-To response header.
+   * Not applicable if viewer can render templates.
    * @param {../../../src/service/xhr-impl.FetchResponse} response
    * @private
    */
   maybeHandleRedirect_(response) {
+    if (this.templateRendererHelper_.canRenderTemplates()) {
+      return;
+    }
```

Comment:
Let's use the error helper here too.

Looks like sometimes we `user().assert`, sometimes we `dev().error`, sometimes we early out. Let's make this behavior consistent -- `dev().error` and early out.